README.md

```diff
@@ -116,7 +116,7 @@ $> clang++ -fsycl –I$ONEMKL/include app.cpp
 $> clang++ -fsycl app.o –L$ONEMKL/lib –lonemkl_blas_mklcpu –lonemkl_blas_cublas
 ```
 
-*Refer to [Selection of Compilers](#selection-of-compilers) for the choice between `dpcpp` and `clang++` compilers.*
+*Refer to [Selecting a Compiler](https://oneapi-src.github.io/oneMKL/selecting_a_compiler.html) for the choice between `dpcpp` and `clang++` compilers.*
```

Comment:
Can you please share the reason for using links instead of Markdown references?
From what I understand, if you are browsing a non-default branch, the new approach will take me to the develop/main branch, instead of the non-default branch's own `README.md`. Is this the intended behavior we want?

Reply:
@vmalia  Good point. I think we should stay in the non-default branch.

Reply:
@mmeterel that is my thought as well. That is why I want to understand the rationale for this change from the documentation team.

Reply:
The links that you're referring to used to go to Markdown content; now those targets are in the reStructuredText content which gets published on GitHub Pages. Therefore, this was not an (intentional) change in approach on my part. I assumed that we'd want to take the reader to the generated documentation. If there is a better way, I will be quite happy to use a reference from the readme to the appropriate reStructuredText file, but the reader won't have the context of the full documentation. Perhaps you could give an example of the syntax that you want. Thanks!

Reply:
I guess, the question is, if the user makes any change to these files/links in their PR, will they have any way to test/verify it in their own branch? I am not fully understanding "publishing on GitHub Pages", but it seems to me that, these changes will not be visible/working until we "publish them on Github Pages". I might be missing something though.

Reply:
The use-cases would look like this:

1. For an end-user: Go to https://oneapi-src.github.io/oneMKL and read the latest info for the latest release.
2. For a contributor: Go to Github and follow markdown docs for the latest "development" documentation.

Reply:
Yes, there is (already) a link to the full documentation earlier in the readme (in fact, a few links to it under Documentation). I could change the later references to be Markdown references. I might ask the InfoDev members who maintain the Intel open-source guidelines what they think about such references. Perhaps then I could submit a fresh pull request to switch those two or three links over. Does that sound good?

Reply:
Yeah, that sounds good to me.

Reply:
If you find that our project is using an unusual approach as compared to other Intel open-source projects, please point that out. Alignment in documentation is important for good customer experience.

Reply:
Great! If you would approve this PR, I'll merge it and then make myself an AR to research the links.

Reply:
Our preference in InfoDev is Jira (Documentation project, for tracking purposes), but either way would be fine.

Reply:
Okay, please send me the JIRA offline.